Xamarin.Forms.Xaml/ApplyPropertiesVisitor.cs

```diff
@@ -36,13 +36,15 @@ public ApplyPropertiesVisitor(HydrationContext context, bool stopOnResourceDicti
 		public bool StopOnResourceDictionary { get; }
 		public bool VisitNodeOnDataTemplate => true;
 		public bool SkipChildren(INode node, INode parentNode) => false;
-		public bool IsResourceDictionary(ElementNode node) => typeof(ResourceDictionary).IsAssignableFrom(Context.Types[node]);
+		public bool IsResourceDictionary(ElementNode node) => Context.Types.TryGetValue(node, out var type) && typeof(ResourceDictionary).IsAssignableFrom(type);
 
 		public void Visit(ValueNode node, INode parentNode)
 		{
 			var parentElement = parentNode as IElementNode;
 			var value = Values [node];
-			var source = Values [parentNode];
+			if (!Values.TryGetValue(parentNode, out var source) && Context.ExceptionHandler != null)
```

Comment:
Isn't adding the `Context.ExceptionHandler != null` changing the behaviour? Is that what we want?

Reply:
we don't want to hide potential bugs, so we only return if the value isn't found AND the previewer is plugged in. If the previewer is not there, we no longer fail with a KeyNotFound, but probably with some NRE later on

none of this should ever happen. if it happens, it indicates a bug in the loader or an error (that we don't handle) in the xaml. In both cases, I want this code to keep failing, the issue to be reported so we can either fix the loader, or throw a meaningful exception